OEP-6/OEP-6.mediawiki

```diff
@@ -0,0 +1,407 @@
+
+===Components===
+Although this proposal is bringing clear and simple API for the dApps, the individual functions can be divided into these components:
+
+* '''Network''', a thin wrapper around the Ontology Node API, masking the complexity of rpc/rest calls and web-sockets with Request-Response facade.
+
+* '''Provider''', functions for getting information about dAPI provider
+
+* '''Asset''', functions for transferring assets between user account and others.
+
+* '''Identity''', functions for interacting with own ONT-ID identity.
+
+* '''SmartContract''', a high level wrapper around the Smart Contract invocation and deployment.
+
+* '''Message''', functions for signing arbitrary messages.
+
+* '''Utils''', a group of utility function for encoding and decoding the data from/to blockchain.
+
+===Network===
+A network API consists of:
+
+<pre>
+type NetworkType = 'MAIN' | 'TEST' | 'PRIVATE';
+type Asset = 'ONT' | 'ONG';
+
+interface Network {
+  type: NetworkType;
+  address: string;
+}
+
+function getGenerateBlockTime(): Promise<number | null>
+function getNodeCount(): Promise<number>
+function getBlockHeight(): Promise<number>
+function getMerkleProof(txHash: string): Promise<MerkleProof>
+function getStorage(contract: string, key: string): Promise<string>
+function getAllowance(asset: Asset, fromAddress: string, toAddress: string): Promise<number>
+function getBlock(block: number | string): Promise<Block>
+function getTransaction(txHash: string): Promise<Transaction>
+function getNetwork(): Network
+function getBalance(address: string): Promise<Balance>
+</pre>
+
+For further explanation about the wrapped method consult https://ontio.github.io/documentation/restful_api_en.html . The types '''Transaction''', '''Block''', '''MerkleProof''' and '''Balance''' corresponds to the exact object returned from Ontology blockchain.
+
+TODO: copy the definition of all network api calls with inputs and outputs from ontology documentation.
+
+===Provider===
+A primary focus of Provider API is to check if user has installed any dAPI provider and to get information about it.
+
+====getProvider====
+<pre>
+
+interface Provider {
+  name: string;
+  version: string;
+  compatibility: string[];
+}
+
+function getProvider(): Promise<Provider>
+</pre>
+
+Returns the information about installed dAPI provider if there is one installed. <code>compatibility</code> refers to all OEPs this dAPI provider supports. E.g.:
+
+<pre>
+compatability: [
+    'OEP-6',
+    'OEP-10',
+    'OEP-29'
+]
+</pre>
+
+dAPI provider does not need to support all future OEPs concerned with dAPI.
+
+* Rejects with '''NO_PROVIDER''' in case there is no dAPI provider installed.
+
+===Asset===
+A primary focus of Asset API is to initiate a transfer. The request needs to be confirmed by the user.
+
+====getDefaultAccount====	
+<pre>	
+function getDefaultAccount(): Promise<string>	
```

Comment:
Was thinking about that. To be orthogonal with getIdentity, I need to make some changes to that part. Because in identity case, one can query also the blockchain for foreign identities and it won't make sense to have getIdentity(identity: string) and getAccount(), where the former one will query the blockchain for arbitrary identity and the second one will get only user account.

Reply:
Where is the `getIdentity(identity: string)` method? It is currently `getDefaultIdentity(): Promise<string>` right?